app/main.py

```diff
@@ -1,21 +1,42 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
-
+    def __init__(self, row, column):
+        self.row = row
+        self.column = column
+        self.is_alive = True
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
+    def __init__(self, start, end):
         # Create decks and save them to a list `self.decks`
-        pass
+        self.start = start
+        self.end = end
+        self.is_drowned = False
+        self.decks = []
+
+    def crate_decks(self, i, j):
+        self.decks.append(Deck(i, j))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    # def get_deck(self, row, column):
+    #     # Find the corresponding deck in the list
+    #     for deck in self.decks:
+    #         if deck.row == row and deck.column == column:
+    #             return deck
 
     def fire(self, row, column):
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        health_ship = []
+
+        for deck in self.decks:
+            if deck.row != row or deck.column != column:
+                health_ship.append(deck.is_alive)
+            elif deck.row == row or deck.column == column:
+                deck.is_alive = False
+                health_ship.append(deck.is_alive)
+        # print(health_ship)
+        if not any(health_ship):
+            self.is_drowned = False
```

Comment:
The logic here is incorrect. If no decks are alive, `self.is_drowned` should be set to `True`, not `False`. This will correctly indicate that the ship is drowned.